src/cpp/core/include/core/http/TcpIpSocketUtils.hpp

```diff
@@ -89,7 +89,10 @@ inline Error initTcpIpAcceptor(
    acceptor.open(endpoint.protocol(), ec);
    if (ec)
       return Error(ec, ERROR_LOCATION);
-   
+
+   // TODO: is this the desired behavior on Windows and desktop?
+   // this could allow multiple clients to try connecting to the
+   // same session
    acceptor.set_option(tcp::acceptor::reuse_address(true), ec);
```

Comment:
@jeffvroom I'd be curious to know your opinion here. Windows has [SO_EXCLUSIVEADDRUSE](https://docs.microsoft.com/en-us/windows/win32/winsock/so-exclusiveaddruse) to disallow multiple applications to bind to the same port, which I _think_ we want here... 

My understanding is that the `reuse_address()` option maps to the `SO_REUSEADDR` socket option, which (for Linux) has documentation, emphasis mine:

> SO_REUSEADDR
>               Indicates that the rules used in validating addresses
>               supplied in a [bind(2)](https://man7.org/linux/man-pages/man2/bind.2.html) call should allow reuse of local
>               addresses.  For AF_INET sockets this means that a socket
>               may bind, **except when there is an active listening socket
>               bound to the address**.  When the listening socket is bound
>               to INADDR_ANY with a specific port then it is not possible
>               to bind to this port for any local address.  Argument is
>               an integer boolean flag.

But on Windows, it means something different (https://docs.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse):

> The SO_REUSEADDR socket option **allows a socket to forcibly bind to a port in use by another socket**. The second socket calls [setsockopt](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-setsockopt) with the optname parameter set to SO_REUSEADDR and the optval parameter set to a boolean value of TRUE before calling [bind](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-bind) on the same port as the original socket. Once the second socket has successfully bound, the behavior for all sockets bound to that port is indeterminate. For example, if all of the sockets on the same port provide TCP service, any incoming TCP connection requests over the port cannot be guaranteed to be handled by the correct socket — the behavior is non-deterministic. A malicious program can use SO_REUSEADDR to forcibly bind sockets already in use for standard network protocol services in order to deny access to those service. No special privileges are required to use this option.

It seems like we'd want the behavior to be uniform here, and so on Windows we either don't want to set this option, or we want to use `SO_EXCLUSIVEADDRUSE` to avoid multiple applications binding on the same port.

Reply:
Nice catch on the meaning of those flags. I totally agree... we should be using SO_EXCLUSIVEADDRUSE on windows and not reuse_address() in boost (that maps to SO_REUSEADDR even on windows).  I found this code snippet:

typedef boost::asio::detail::socket_option::boolean<BOOST_ASIO_OS_DEF(SOL_SOCKET), SO_EXCLUSIVEADDRUSE> excluse_address;  acceptor.set_option(excluse_address(true));

in a comment on this page:

https://stackoverflow.com/questions/7164879/boost-asio-why-dont-i-get-bind-address-already-in-use-in-windows-but-do-ge